sponsorship_compassion/model/contracts.py

```diff
@@ -749,61 +700,56 @@ def _has_valid_contract_lines(
                         str(allowed)) if allowed else _(
                         'You should not select product '
                         'from category "{0}"'.format(categ_name))
-                    raise orm.except_orm(
+                    raise exceptions.Warning(
                         _('Please select a valid product'), message)
         return True
 
-    def update_next_invoice_date(self, cr, uid, ids, context=None):
+    @api.multi
+    def update_next_invoice_date(self):
         """ Override to force recurring_value to 1
             if contract is a sponsorship, and to bypass ORM for performance.
         """
-        group_ids = list()
-        for contract in self.browse(cr, uid, ids, context):
-            group_ids.append(contract.group_id.id)
+        groups = self.mapped('group_id')
+        for contract in self:
             if 'S' in contract.type:
                 next_date = datetime.strptime(contract.next_invoice_date, DF)
                 next_date += relativedelta(months=+1)
                 next_date = next_date.strftime(DF)
             else:
-                next_date = self._compute_next_invoice_date(contract)
+                next_date = contract._compute_next_invoice_date()
 
-            cr.execute(
+            self.env.cr.execute(
                 "UPDATE recurring_contract SET next_invoice_date = %s "
                 "WHERE id = %s", (next_date, contract.id))
-        self.pool.get('recurring.contract.group')._store_set_values(
-            cr, uid, group_ids, ['next_invoice_date'], context)
-
+        groups._store_set_values(['next_invoice_date'])
         return True
 
-    def _get_filtered_invoice_lines(
-            self, cr, uid, invoice_lines, contract_id, context=None):
-        res = list()
-        for invl in invoice_lines:
-            if (invl.contract_id == contract_id and
-               invl.product_id.categ_name != GIFT_CATEGORY):
-                res.append(invl.id)
+    @api.multi
+    def _get_filtered_invoice_lines(self, invoice_lines, contract_id):

```

Comment:
Why contract_id is needed ? Should it be `self.id` instead with api.one or `self.ids` with api.multi ?